File: server/http/Model/SystemMailUser.cs
using StackExchange.Redis;

namespace Http.Model
{
    public class SystemMailUserKey : BaseModel, IRedisHashKey
    {
        public required uint User { get; set; }
        public required uint MailId { get; set; }
        public uint? GetHash() => User;

        public RedisKey GetRedisKey() => $"fb:cache:system_mail_user:{User}";

        public RedisValue GetRedisField() => MailId.ToString();
    }

    public class SystemMailUser : SystemMailUserKey, IModel
    {
        public bool Read { get; set; }
        public DateTime? ExpireDate { get; set; }
    }
}

File: server/game/lib/system_mail_user.cpp
#include <fb/game/system_mail_user.h>

using namespace fb::game;

system_mail_user::system_mail_user(uint32_t mail_id, bool read, const std::optional<std::string>& expire_date) :
    mail_id(mail_id),
    read(read),
    expire_date(expire_date.has_value() ? std::make_optional<fb::model::datetime>(expire_date.value()) : std::nullopt)
{ }
